tracing/tests/support/collector.rs

```diff
@@ -520,6 +520,10 @@ where
             }
         }
     }
+
+    fn current_span(&self) -> tracing_core::span::Current {
+        todo!()
+    }
```

Comment:
Why's this one todo'd, whereas all the others are unknown?

Reply:
the test collector probably _should_ actually implement this, it knows the ID of the current span:
https://github.com/tokio-rs/tracing/blob/f867e83dbaaa9ae73d7ee26a6766bb9b49730d06/tracing/tests/support/collector.rs#L44
and it can look up spans by ID:
https://github.com/tokio-rs/tracing/blob/f867e83dbaaa9ae73d7ee26a6766bb9b49730d06/tracing/tests/support/collector.rs#L42

similar to the `sloggish` example collector, we would need to add the metadata to the stored span data to implement this properly. in this case, the stored span data is the `SpanState` struct:
https://github.com/tokio-rs/tracing/blob/f867e83dbaaa9ae73d7ee26a6766bb9b49730d06/tracing/tests/support/collector.rs#L36-L39

we could pretty easily add a `&'static Metadata<'static>` to that, allowing us to actually return the current span.

i think this is somewhat important; even though none of the tests currently use the mock collector's `current_span` method, it should probably implement the behavior correctly so that we're not surprised when writing future tests that expect it to work...